python/taichi/lang/kernel_impl.py

```diff
@@ -656,29 +656,35 @@ def func__(*args):
                     is_numpy = isinstance(v, np.ndarray)
                     is_torch = isinstance(v,
                                           torch.Tensor) if has_torch else False
+                    is_soa = needed.layout == Layout.SOA
+                    array_shape = v.shape
+                    element_dim = needed.element_dim
```

Comment:
This is not super straightforward for readers ;) - shall we add a note here explaining that we are viewing an external array with specified element_dim etc?